runbotics/runbotics-scheduler/src/websocket/gateway/bot.gateway.ts

```diff
@@ -39,14 +40,25 @@ export class BotWebSocketGateway implements OnGatewayDisconnect, OnGatewayConnec
 
         const { bot } = await this.authService.validateBotWebsocketConnection({ client });
         this.uiGateway.server.emit(WsMessage.BOT_STATUS, bot);
-
         this.logger.log(`Bot connected: ${bot.installationId} | ${client.id}`);
         client.join(bot.installationId);
+        const processInstanceEventObservable$ = fromEvent(client, BotWsMessage.PROCESS_INSTANCE_EVENT).pipe(
+            filter((processInstanceEvent: IProcessInstanceEvent) => 
+                processInstanceEvent.status === ProcessInstanceEventStatus.COMPLETED || processInstanceEvent.status === ProcessInstanceEventStatus.IN_PROGRESS
+            ),
+            delayWhen((event) => {
+                if (event.status === ProcessInstanceEventStatus.COMPLETED) {
+                    return of(null);
+                }
+                return timer(500);
+            }),
+        );
+
+        this.listenForProcessInstanceEvent(bot, processInstanceEventObservable$);
```

Comment:
I prefer to move this logic to the service. Stasiu created one that covers a similar sense, could you leave him a message so he moves this during conflict resolution?